clang/test/SemaSYCL/attr-syclglobalvar.cpp

```diff
@@ -0,0 +1,54 @@
+// RUN: %clang_cc1 -fsycl-is-device -verify -fsyntax-only %s
+
+#include "Inputs/sycl.hpp"
+
+__attribute__((sycl_global_var)) int GlobalWithAttribute;
+
+__attribute__((sycl_global_var)) extern int ExternGlobalWithAttribute;
+
+namespace NS {
+    __attribute__((sycl_global_var)) int NSGlobalWithAttribute;
+}
+
+union U {
+    int InstanceMember;
+};
+
+__attribute__((sycl_global_var)) U GlobalUnion;
```

Comment:
I was thinking of a slightly different test here:
```
static union {
__attribute__((sycl_global_var)) int InstanceMember;
};
```

Reply:
Is static on a type name allowed? I get a warning diagnostic: `'static' is not permitted on a declaration of a type`

Reply:
https://godbolt.org/z/b4b1cM9f8

Reply:
I see. I added the anonymous static union to the tests. The anonymous union, it turns out, is not considered a global (doesn't accept the attribute) *and* isn't allowed in a kernel.